internal/slices/iter.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package slices
+
+import (
+	"iter"
+)
+
+// AppliedToEach returns an iterator that yields the slice elements transformed by the function `f`.
+func AppliedToEach[S ~[]E, E any, T any](s S, f func(E) T) iter.Seq[T] {
+	return func(yield func(T) bool) {
+		for _, v := range s {
+			if !yield(f(v)) {
+				return
+			}
+		}
+	}
+}
+
+// Backward returns an iterator that yields the slice elements in reverse order.
+func BackwardValues[Slice ~[]E, E any](s Slice) iter.Seq[E] {
```

Comment:
Rename to `ReverseValues` to match `slices.Reverse`?

Reply:
It's a values-only equivalent of [`slices.Backward`](https://cs.opensource.google/go/go/+/go1.24.5:src/slices/iter.go;l=26)